pkg/opentelemetry/controller.go

```diff
@@ -98,35 +77,23 @@ func (c *Controller) convertTime(t int64) time.Time {
 }
 
 // NewController returns a new initialized [Controller].
-func NewController() (*Controller, error) {
-	serviceName, exists := os.LookupEnv(otelServiceNameEnvVar)
-	if !exists {
-		return nil, fmt.Errorf("%s env var must be set", otelServiceNameEnvVar)
-	}
-
-	ctx := context.Background()
+func NewController(
+	ctx context.Context,
+	serviceName string,
+	exporter sdktrace.SpanExporter,
```

Comment:
not a blocker yet since this isn't stable but we might want to consider a different approach to passing params in if this list could grow

Reply:
Updated . Added controller settings.